src/common/services/api/cart.service.js

```diff
@@ -82,9 +87,36 @@ class Cart {
       });
   }
 
-  addItem(uri, data){
+  getTotalQuantity() {
+    return this.cortexApiService
+      .get({path: ['carts', this.cortexApiService.scope, 'default']})
+      .map((cart) => {
+        return cart['total-quantity'];
+      });
+  }
+
+  addItem(uri, data) {
     data.quantity = 1;
 
+    if(this.sessionService.getRole() == Roles.public) {
```

Comment:
This might be cleaner in a separate function. That would clarify what it does and might help with the duplicate calls.